src/platform/ESP32/bluedroid/BLEManagerImpl.cpp

```diff
@@ -277,21 +336,447 @@ void BLEManagerImpl::_OnPlatformEvent(const ChipDeviceEvent * event)
+                                                        gl_profile_tab[PROFILE_A_APP_ID].service_start_handle,
+                                                        gl_profile_tab[PROFILE_A_APP_ID].service_end_handle, char_elem_result,
+                                                        &count, offset);
+                    if (status != 0)
+                    {
+                        ChipLogProgress(Ble, "esp_ble_gattc_get_char_by_uuid error");
+                    }
+
+                    /*  Every service have only one char in our 'ESP_GATTS_DEMO' demo, so we used first 'char_elem_result' */
+                }
+            }
+            if (count > 0)
+            {
+
+                for (int i = 0; i < count; i++)
+                {
+                    if (char_elem_result[i].uuid.len == ESP_UUID_LEN_128)
+                    {
+                        if (char_elem_result[i].properties & CharProps_Write)
+                        {
+                            gl_profile_tab[PROFILE_A_APP_ID].write_char_handle = char_elem_result[i].char_handle;
+                        }
+                        else if (char_elem_result[i].properties & CharProps_ReadNotify)
+                        {
+                            gl_profile_tab[PROFILE_A_APP_ID].notify_char_handle = char_elem_result[i].char_handle;
+                            esp_ble_gattc_register_for_notify(gl_profile_tab[PROFILE_A_APP_ID].gattc_if,
+                                                              gl_profile_tab[PROFILE_A_APP_ID].remote_bda,
+                                                              char_elem_result[i].char_handle);
+                        }
+                    }
+                }
+            }
+            free(char_elem_result);
+        }
+        else
+        {
+            ChipLogProgress(Ble, "no char found");
+        }
+    }
+    break;
+    case ESP_GATTC_REG_FOR_NOTIFY_EVT: {
+        ChipLogProgress(Ble, "ESP_GATTC_REG_FOR_NOTIFY_EVT");
+        if (p_data->reg_for_notify.status != ESP_GATT_OK)
+        {
+            ChipLogProgress(Ble, "REG FOR NOTIFY failed: error status = %d", p_data->reg_for_notify.status);
+        }
+        else
+        {
+            uint16_t count               = 0;
+            esp_gatt_status_t ret_status = esp_ble_gattc_get_attr_count(
+                gl_profile_tab[PROFILE_A_APP_ID].gattc_if, gl_profile_tab[PROFILE_A_APP_ID].conn_id, ESP_GATT_DB_DESCRIPTOR,
+                gl_profile_tab[PROFILE_A_APP_ID].service_start_handle, gl_profile_tab[PROFILE_A_APP_ID].service_end_handle,
+                gl_profile_tab[PROFILE_A_APP_ID].notify_char_handle, &count);
+            if (ret_status != ESP_GATT_OK)
+            {
+                ChipLogProgress(Ble, "esp_ble_gattc_get_attr_count error");
+            }
+            if (count > 0)
+            {
+                descr_elem_result = (esp_gattc_descr_elem_t *) malloc(sizeof(esp_gattc_descr_elem_t) * count);
```

Comment:
I don't see a call to `free` for this, is that expected?

Reply:
yes, It should be freed , I will add this change